indexer/ai/src/main/java/com/google/cloudsearch/ai/StandardSkillCategoryExtraction.java

```diff
@@ -0,0 +1,198 @@
+     */
+    @Override
+    public void setFilter(JSONObject filter) {
+        this.categoryConfidence = (double) filter.get(Constants.CONFIG_CATEGORY_CONFIDENCE);
+    }
+
+    /**
+     * Returns the filters
+     * @return  Returns the filters in JSON Format.
+     */
+    @Override
+    public JSONObject getFilter() {
+        JSONObject obj = new JSONObject();
+        obj.put(Constants.CONFIG_CATEGORY_CONFIDENCE, this.categoryConfidence);
+        return obj;
+    }
+
+  /**
+   * Parse the JSON Object containing the filters for the skill and validate it before calling the setter.
+   * @param filter  JSON Object containing the filters
+   * @throws InvalidConfigException     Throws exception if the filter name is not supported.
+   */
+  @Override
+  public void parseFilter(JSONObject filter) throws InvalidConfigException {
+        if(filter == null) {
+            return;
+        }
+        for(Object key : filter.keySet()) {
+            if (key.equals(Constants.CONFIG_CATEGORY_CONFIDENCE))
+                setFilter(filter);
+            else {
+                throw new InvalidConfigException("Filter " + key + " is not supported for AISkill Category Extraction.");
+            }
+        }
+    }
+
+    /**
+     * Check if the confidence score satisfied the specified filter.
+     * @param confidence    Confidence score obtained for each category returned by the API
+     * @return              Returns true if the condition is satisfied, else returns false.
+     */
+    private boolean isFilterSatisfied(double confidence) {
+        return (confidence >= this.categoryConfidence);
+    }
+
+    /**
+     * Parses the Configuration for the Category Extraction Skill
+     * @param aiSkill        JSONObject for the skill
+     * @param schema         CloudSearch schema
+     * @throws InvalidConfigException
+     */
+    @Override
+    protected void parse(JSONObject aiSkill, JSONObject schema) throws InvalidConfigException {
+        parseAISkillName((String) aiSkill.get(Constants.CONFIG_SKILL_NAME));
+        parseOutputMappings((JSONArray) aiSkill.get(Constants.CONFIG_OUTPUT_MAPPINGS), schema);
+        parseInputs((JSONObject) aiSkill.get(Constants.CONFIG_INPUTS));
+        parseFilter((JSONObject) aiSkill.get(Constants.CONFIG_FILTERS));
+    }
+
+    /**
+     * Set up the language service client.
+     * @throws Exception    Throws IOException if client creation fails.
+     */
+    @Override
+    public void setupSkill() throws Exception {
+        languageService = LanguageServiceClient.create();
+    }
+
+    /**
+     * Execute the Category Extraction Skill and populates the structured Data.
+     * @param contentOrURI      The actual file content or Cloud storage URI
+     * @param structuredData    Multimap to store the structured data for indexing.
+     */
+    @Override
+    public void executeSkill(String contentOrURI, Multimap<String, Object> structuredData) {
```

Comment:
nit: maybe add a check if setup was actually called.

Reply:
languageService will remain null in that case, and the try block will throw null pointer exception, since I am anyway the catch statement is for any exception it should be fine. Though I can also add an explicit check and throw a null pointer exception.

Reply:
Null Pointer Exceptions are rarely of any help, its always advised to avoid them.

Reply:
Sure, IllegalStateException with the message language client not initialized or setup not called.

Reply:
Done